src/Parallel/Callback.hpp

```diff
@@ -0,0 +1,95 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines Parallel::Callback.
+
+#pragma once
+
+#include <pup.h>
+#include <tuple>
+#include <utility>
+
+#include "Parallel/CharmPupable.hpp"
+#include "Parallel/Invoke.hpp"
+
+namespace Parallel {
+/// A class that holds a function that can be invoked at a later time.
```

Comment:
Might be helpful to specify in the dox that this is an abstract base class, just because that's a feature that's not necessarily obvious in the doxygen interface otherwise.